brainscore/submission/evaluation.py

```
@@ -102,48 +121,69 @@ def run_evaluation(config_file, work_dir, jenkins_id, db_secret, models=None,
-    return candidates[0]
-
-
-def clone_repo(config, work_dir):
-    git.Git(work_dir).clone(config['git_url'])
-    return Path('%s/%s' % (work_dir, os.listdir(work_dir)[0]))
-
-
-def install_project(repo, package):
-    try:
-        assert 0 == subprocess.call([sys.executable, "-m", "pip", "install", "-v", repo], env=os.environ)
-        sys.path.insert(1, str(repo))
-        logger.info(f'System paths {sys.path}')
-        return import_module(package)
-    except ImportError:
-        return __import__(package)
+def get_ml_pool(test_models, module, submission):
+    ml_brain_pool = {}
+    if submission.model_type == 'BaseModel':
+        logger.info(f"Start working with base models")
+        layers = {}
+        base_model_pool = {}
+        for model in test_models:
+            function = lambda model_inst=model.name: module.get_model(model_inst)
+            base_model_pool[model.name] = LazyLoad(function)
+            try:
+                layers[model.name] = module.get_layers(model.name)
+            except Exception:
+                logging.warning(f'Could not retrieve layer for model {model} -- skipping model')
+        model_layers = ModelLayers(layers)
+        ml_brain_pool = MLBrainPool(base_model_pool, model_layers)
+    else:
+        logger.info(f"Start working with brain models")
+        for model in test_models:
+            ml_brain_pool[model.name] = module.get_model(model.name)
+    return ml_brain_pool
+
+
+def get_benchmark_instance(benchmark):
+    bench = benchmark_pool[benchmark]
+    benchmark_type, created = BenchmarkType.get_or_create(identifier=benchmark, order=999)
+    if created:
+        try:
+            parent = BenchmarkType.get(identifier=bench.parent)
+            benchmark_type.parent = parent
+            benchmark_type.save()
+        except DoesNotExist:
+            logger.error(
+                f'Couldn\'t connect benchmark {benchmark} to parent {bench.parent} since parent doesn\'t exist')
```

I would either raise an error here or turn this into a warning